lib/raw_net_capture.rb

```diff
@@ -9,14 +9,22 @@ def initialize
   end
 
   def received(data)
+    @received = true
```

Comment:
I still don't think there's any need to change `RawNetCapture`, it should just be a version of the `<-` `->` stuff you usually get from `debug_output` but in a programmatically-accessible way ... it could be used to tap into any raw socket-level communication, not just net/http stuff